README.md

```diff
@@ -0,0 +1,132 @@
+# WORK IN PROGRESS
+
+This library is still under development and hence subject to breaking changes.
+
+## Summary
+
+This is a Go library used to connect to a [HashiCorp Consul](https://www.consul.io/) server. It
+implements server discovery and provides a gRPC connection to a Consul server.
+
+It was designed to maintain a current connection to a Consul server in absence of a Consul client
+agent.
+
+It supports the following:
+
+* Discovering Consul server addresses using
+  [go-netaddrs](https://github.com/hashicorp/go-netaddrs) and Consul's [ServerWatch gRPC
+  stream](https://github.com/hashicorp/consul/blob/main/proto-public/pbserverdiscovery/serverdiscovery.proto)
+* Connecting to a Consul server over gRPC
+* Automatic rediscovery and reconnection to another Consul server
+* Consul ACL token authentication
+* Compatibility with Consul server xDS load balancing
```

Comment:
Also supports optional custom server filtering.

Reply:
I think I'll make another PR to add a couple of other sections for a server filtering example, and for the "server watch disabled / load balancer" case.